packages/modules/clickhouse/src/clickhouse-container.test.ts

```diff
@@ -0,0 +1,102 @@
+import { ClickhouseContainer, StartedClickhouseContainer } from "./clickhouse-container";
+import { ClickHouseClient, createClient } from "@clickhouse/client";
+import path from "path";
+
+describe("ClickhouseContainer", () => {
+  jest.setTimeout(240_000);
+
+  it("should work with defaults", async () => {
+    const container = await new ClickhouseContainer().start();
+    const client = createClickhouseContainerHttpClient(container);
+    await _test(client, container.getDatabase());
+    await client.close();
+    await container.stop();
+  });
+
+  it("should work with custom credentials", async () => {
+    const container = await new ClickhouseContainer()
+      .withUsername(`un${(Math.random()*1000000) | 0}`)
+      .withPassword(`pass${(Math.random()*1000000) | 0}`)
+      .start();
+    const client = createClickhouseContainerHttpClient(container);
+    await _test(client, container.getDatabase());
+    await client.close();
+    await container.stop();
+  });
+
+  it("should work with custom database", async () => {
+    const container = await new ClickhouseContainer()
+      .withDatabase(`db${(Math.random()*1000000) | 0}`)
+      .start();
+    const client = createClickhouseContainerHttpClient(container);
+    await _test(client, container.getDatabase());
+    await client.close();
+    await container.stop();
+  });
+
+  it("should work with custom image", async () => {
+    const container = await new ClickhouseContainer("23.8-alpine").start();
+    const client = createClickhouseContainerHttpClient(container);
+    await _test(client, container.getDatabase());
+    await client.close();
+    await container.stop();
+  });
+
+  it("should work with custom xml config", async () => {
+    const container = await new ClickhouseContainer()
+      .withPassword("")
+      .withXmlConfigFile(path.join("testdata", "config.xml"))
+      .start();
+    const client = createClickhouseContainerHttpClient(container);
+    await _test(client, container.getDatabase());
+    await client.close();
+    await container.stop();
+  });
+
+  it("should work with custom yaml config", async () => {
+    const container = await new ClickhouseContainer()
+      .withPassword("")
+      .withXmlConfigFile(path.join("testdata", "config.yaml"))
+      .start();
+    const client = createClickhouseContainerHttpClient(container);
+    await _test(client, container.getDatabase());
+    await client.close();
+    await container.stop();
+  });
+  
+  function createClickhouseContainerHttpClient(container: StartedClickhouseContainer) {
+    return createClient({
+      host: `http://${container.getHost()}:${container.getHostPorts().http}`,
```

Comment:
is this should be returned by the container implementation for easy access?

Reply:
I think the client should be a different module rather than part the container abstraction
it also couples the dev to a dep he may not want (even though it may make sense to couple because its an official client), for example qryn project doesnt use this client atm